src/fmtnum.rs

```diff
@@ -0,0 +1,121 @@
+macro_rules! fmtint {
+    ($($t:ident)*) => ($(
+        #[allow(unused_comparisons)]
+        impl<'a, 'b> Formatter<'a, 'b> {
+            pub fn $t(&mut self, x: $t) -> Result<()> {
+                let ty = match self.ty() {
+                    None => ' ',
+                    Some(c) => c,
+                };
+
+                if !self.is_int_type() {
+                    let mut msg = String::new();
+                    write!(msg, "Unknown format code {:?} for type", ty).unwrap();
+                    return Err(FmtError::TypeError(msg));
+                }
+
+                if self.precision() != None {
+                    return Err(FmtError::TypeError("precision not allowed for integers".to_string()));
+                }
+
+                if self.thousands() {
+                    return Err(FmtError::Invalid("thousands specifier not yet supported".to_string()));
+                }
+
+                if self.fill() == '0' && self.align() == Alignment::Right {
+                    return Err(FmtError::Invalid("sign aware 0 padding not yet supported".to_string()));
+                }
+
+                let mut s = String::new();
+
+                if x >= 0 && self.sign_plus() {
+                    self.write_str("+").unwrap();
+                }
+
+                if self.alternate() {
+                    match ty {
+                        'b' => self.write_str("0b").unwrap(),
+                        'o' => self.write_str("0o").unwrap(),
+                        'x' | 'X' => self.write_str("0x").unwrap(),
+                        _ => {
+                            let mut msg = String::new();
+                            write!(msg, "alternate ('#') cannot be used with type {:?}", ty).unwrap();
```

Comment:
when I first wrote this lib I didn't know about `format!()`. Please use that instead here. You can even get rid of `msg` variable alltogether.

Reply:
whoops, I'll handle it in a separate commit... maybe (who really cares, haha).

Thanks for the PR!